### PR TITLE
Tooltip V0.2 with portals

### PR DESCRIPTION
**Pull request by dancingshell**
### Summary

* new API with portals!
* tooltips can now be controlled (not hoverable, show prop passed in by parent) allowing it to stay open

### Screenshots
awesomescreenshot-2017-11-17t19-11-32-172z

### CODE REVIEW

File: tooltip/src/Tooltip.jsx [hunk 1]
```diff
@@ -1,130 +1,126 @@
...
-  }
+    controlled: false,
+    placement: 'top',
+    show: false,
+    title: ''
+  };
+
+  static propTypes = {
+    controlled: PropTypes.bool,
+    placement: PropTypes.oneOf(['top', 'bottom', 'left', 'right']).isRequired,
```
Comment (ben-burlingham) [inline tooltip/src/Tooltip.jsx]: nb Could these be expressed as an enum, and exported?

File: tooltip/src/Tooltip.jsx [hunk 2]
```diff
@@ -1,130 +1,126 @@
...
+    show: PropTypes.bool,
+    title: PropTypes.string.isRequired
```
Comment (mcclayton) [inline tooltip/src/Tooltip.jsx]: `nb` I'd consider generalizing this to something like `content`. This gives us the flexibility to turn this into `Proptypes.node` in the future without needing a breaking change and makes it a little more general.

File: tooltip/src/Tooltip.jsx [hunk 3]
```diff
@@ -1,130 +1,126 @@
...
-import composeStyles from '../../shared/stylesheetComposer';
 
-export default class Tooltip extends React.Component {
-  static propTypes = {
-    stylesheets: PropTypes.arrayOf(PropTypes.shape())
-  }
+import TooltipPortal from './TooltipPortal';
+import styles from './Tooltip.scss';
 
+const ADJUSTMENT = 8;
```
Comment (mcclayton) [inline tooltip/src/Tooltip.jsx]: Is this the padding between the tooltip pointer and the component it's wrapping?
Might be nice to add a comment if so.

File: tooltip/src/Tooltip.jsx [hunk 4]
```diff
@@ -1,130 +1,126 @@
...
+    if (this.state.show && prevState.show === false) {
+      this.position();
+    }
   }
 
-  hideTooltip = () => {
-    this.setState({ title: '' });
-  }
+  onMouseEnter = () => {
+    if (!this.props.controlled) {
```
Comment (mcclayton) [inline tooltip/src/Tooltip.jsx]: I love the idea of being able to control the tooltip from a parent or have the state self-managed.
In the future I could easily see the self-managed tooltip state being managed by a HoC. This seems to be equally great though :)

File: tooltip/src/Tooltip.jsx [hunk 5]
```diff
@@ -1,130 +1,126 @@
...
-        break;
+  onMouseLeave = () => {
+    if (!this.props.controlled) {
+      this.setState({ show: false });
+    }
+  };
+
+  position() {
+    if (this.parentElement && this.tooltip) {
+      const {
```
Comment (mcclayton) [inline tooltip/src/Tooltip.jsx]: `nb`: Not blocking - just pointing out that not all of these properties are available to older browsers:
Here's an example of what IE 8 vs. 9 returns -> https://stackoverflow.com/a/17686282
Not blocking though.

**Comment by mcclayton** [comment]
You'll likely want a throttled window resize handler to help reposition the tooltip on window resize. This will be evident for controlled tooltips.

**Comment by mcclayton** [comment]
Final thing I'd call out (all of my comments are non-blocking)
is that the styling is slightly different from what @nickwmurphy had.
screen shot 2017-11-17 at 11 55 20 am
screen shot 2017-11-17 at 11 56 00 am

This may be intentional, but calling out here in case it wasn't.

File: tooltip/src/Tooltip.jsx (hunk 3 above)
Comment (dancingshell) [reply tooltip/src/Tooltip.jsx]: Removed and re-implemented with css

**Comment by dancingshell** [comment]
@mcclayton @nickwmurphy @ben-burlingham 
Updates
- Added `popover` prop that applies popover styling
- Export position options
- Controlled tooltips now reposition on page resize
- Implemented old adjusted position var with css

File: tooltip/src/Tooltip.jsx [hunk 6]
```diff
@@ -63,7 +85,17 @@ class TooltipWrapper extends React.Component {
...
+    if (!this.throttleTimeout) {
+      this.throttleTimeout = setTimeout(() => {
+        this.throttleTimeout = null;
+        this.position();
+      }, 66);
+    }
+  }
+
   position() {
+    console.log('trying')
```
Comment (mcclayton) [inline tooltip/src/Tooltip.jsx]: ^ console

Comment (dancingshell) [reply tooltip/src/Tooltip.jsx]: dang 🙃

**Comment by dancingshell** [comment]
also here is the popover styles 
popover

**Pull request merged**